data/sql/datasource/datasource/src/main/java/io/helidon/data/sql/datasource/ProviderConfig.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024, 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.data.sql.datasource;
+
+import io.helidon.common.config.NamedService;
+// DataSourceProviderConfig
+
+/**
+ * Provider specific configuration.
```

Comment:
What rules does this interface add to `NamedService`? If the answer is "none", why does this interface exist? If the answer is not "none", then describe them here.

Reply:
@tomas-langer Those config blueprints is your concept.

Reply:
When used, we want to only permit `ProviderConfig` instances, not any `NamedService` out there.